### ISSUE TITLE
Modernize code to not use legacy functions

### ISSUE THREAD

**Issue by pavelkomarov** [pavelkomarov/exportify#48]
`React.createElement` is [apparently no longer favored](https://react.dev/reference/react/createElement). Given how verbose it makes the code, I'm not surprised, but I don't really know the modern alternative. You can `document.createElement`, but that's even more verbose because [element creation and property/children-setting have to be done on separate lines](https://eloquentjavascript.net/14_dom.html#h-XgjABY6Ugx). You could set inner html directly, making clever use of `display=none`. Unclear what the best course is, but my formerly minimal, easily-understood code is getting too long for my taste and needs some help.

**Comment by pavelkomarov** [comment]
https://react.dev/learn/writing-markup-with-jsx

**Comment by pavelkomarov** [comment]
If somebody out there knows JSX and wants to submit a quick PR, I'd be grateful. Otherwise this is my opportunity to (eventually) try JSX directly.

**Comment by ChinoUkaegbu** [comment]
Would love to help on this!

**Comment by pavelkomarov** [comment]
I don't use a particular formatter. I laid out all the code files manually.

**Comment by ChinoUkaegbu** [comment]
**TLDR: Using `React.createElement` is [fine](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#:~:text=If%20you%20need%20to%20manually%20create%20elements%20in%20your%20code%2C%20you%20should%20keep%20using%20React.createElement.%20It%20will%20continue%20to%20work%20and%20is%20not%20going%20away.)**

While it has in fact been moved to legacy APIs, using `React.createElement` seems to still be supported. JSX is the preferred format because it is more readable and similar to HTML but in order to use JSX, it has to be transpiled into JS under the hood (since most browsers do not directly understand JSX). This transformation is typically done with some compiler like Babel or Create React App. Previously, these compilers would take our JSX and transform it to JS by manually calling `React.createElement` under the hood. But now the compilers transform the JSX into some more performance-friendly [format](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-different-in-the-new-transform) but support for `React.createElement` is still being retained.

Essentially, we could still rewrite the code and use JSX and the compilers and all that, but to me that seems unnecessary since it's not going to be deprecated anyways. Plus if we do set up the compiler, we're going to get a JS file either way after the transformation that would be slightly faster but...idk doesn't really seem worth the hassle. I've rewritten the code in JSX though so if you still want to go ahead with it, I'm on board (it would introduce a couple more dependencies though since we need to install the compiler package with npm or use Create React App which I'm not too familiar with but seems basic enough to set up).

And this just reminded me that I never got round to finishing Silicon Valley, so now I've got something to do over the weekend lol

**Comment by pavelkomarov** [comment]
I would like to see your JSX version if you've already done it. You keep the javascript file in the repo, or the jsx in the repo, or both?
